lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -204,21 +214,18 @@ def safe_add_concurrent_partitioned_index(
       PgHaMigrations::Index.from_table_and_columns(child_table, columns)
     end
 
-    # TODO: take out ShareLock after issue #39 is implemented
-    safely_acquire_lock_for_table(parent_table.fully_qualified_name) do
-      # CREATE INDEX ON ONLY parent_table
-      unsafe_add_index(
-        parent_table.fully_qualified_name,
-        columns,
-        name: parent_index.name,
-        if_not_exists: if_not_exists,
-        using: using,
-        unique: unique,
-        where: where,
-        comment: comment,
-        algorithm: :only, # see lib/pg_ha_migrations/hacks/add_index_on_only.rb
-      )
-    end
+    # CREATE INDEX ON ONLY parent_table
+    unsafe_add_index(
+      parent_table.fully_qualified_name,
+      columns,
+      name: parent_index.name,
+      if_not_exists: if_not_exists,
+      using: using,
+      unique: unique,
+      where: where,
+      comment: comment,
+      algorithm: :only, # see lib/pg_ha_migrations/hacks/add_index_on_only.rb
+    )
```

Comment:
This changed because locking concerns have been moved to the `unsafe_add_index` method itself

Reply:
With the latest commit  I'm doing locking directly in the `safe_add_index_on_empty_table` method, so I backed this change out for now. I think we can more holistically solve explicit locking for various generic methods in a different PR